### PR TITLE
fix(rds): add clusterResourceIdentifier property to database cluster

### PR DESCRIPTION
**Pull request by alexedwardjones**
**Summary**

- Exposed the [`DBClusterResourceId`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html#aws-resource-rds-dbcluster-return-values) on the database cluster as `clusterResourceIdentifier`.
- This is needed to [create an IAM policy for IAM database access](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/UsingWithRDS.IAMDBAuth.IAMPolicy.html).

### CODE REVIEW

**Comment by alexedwardjones** [comment]
Is anybody able to review this?

**Comment by alexedwardjones** [comment]
@TheRealAmazonKendra @rix0rrr Any chance of a review on this?

File: packages/@aws-cdk/aws-rds/lib/cluster-ref.ts [hunk 1]
```diff
@@ -14,6 +14,13 @@ export interface IDatabaseCluster extends IResource, ec2.IConnectable, secretsma
    */
   readonly clusterIdentifier: string;
 
+  /**
+   * The immutable identifier for the cluster; for example: cluster-ABCD1234EFGH5678IJKL90MNOP.
+   *
+   * This AWS Region-unique identifier is used in things like IAM authentication policies.
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-rds/lib/cluster-ref.ts:20]:
```suggestion
   * This AWS Region-unique identifier is used to grant access to the cluster.
   *
   * @see https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/UsingWithRDS.IAMDBAuth.IAMPolicy.html
```

**Review by comcalvi — CHANGES_REQUESTED** [review]
do we need to expose this identifier? It seems to just be a component of the arn, eg

```
arn:aws:rds-db:region:account-id:dbuser:DbClusterResourceId/db-user-name
```

If the only goal is to refer to the cluster in IAM policies, the arn being exposed should be sufficient here, since the `DbClusterResourceId` is only used to construct the ARN. If we absolutely must use this `DbClusterResourceId`, can we hide that complexity behind some`grant*()` methods.

File: packages/@aws-cdk/aws-rds/lib/cluster-ref.ts (hunk 1 above)
Comment (alexedwardjones) [reply packages/@aws-cdk/aws-rds/lib/cluster-ref.ts:20]: Hey @comcalvi, thanks for the review 😄 

Unfortunately the ARN isn't what we need here. The cluster identifier and resource identifier are different.

**RDS Cluster ARN:** arn:aws:rds:_region_:_account_:_clusterIdentifier_
**RDS Cluster IAM Auth Access ARN:** arn:aws:rds:_region_:_account_:dbuser:_clusterResourceIdentifier_/_databaseUser_

I was following the existing examples of [docdb](https://github.com/aws/aws-cdk/blob/74318c7d22bfc00de9e005f68a0a6aaa58c7db39/packages/%40aws-cdk/aws-docdb/lib/cluster.ts#L324-L344) and [neptune](https://github.com/aws/aws-cdk/blob/74318c7d22bfc00de9e005f68a0a6aaa58c7db39/packages/%40aws-cdk/aws-neptune/lib/cluster.ts#L243-L252) that expose both the cluster identifier and cluster resource identifiers as properties.

**Review by comcalvi — CHANGES_REQUESTED** [review]

**Review by comcalvi — APPROVED** [review]

**Pull request merged**